lib/pod/command/trunk/push.rb

```diff
@@ -64,14 +64,20 @@ def run
           validate_podspec
           json = push_to_trunk
           update_master_repo
-          print_messages(json['data_url'], json['messages'])
+
+          if (400...600).cover?(@push_status)
+            print_messages(json['data_url'], nil, nil)
+          else
+            print_messages(json['data_url'], json['messages'], spec, 'published')
+          end
         end
 
         private
 
         def push_to_trunk
           response = request_path(:post, "pods?allow_warnings=#{@allow_warnings}",
                                   spec.to_json, auth_headers)
+          @push_status = response.status_code
```

Comment:
maybe return both the push status and the json for the redirected url in this method, instead of setting the ivar?